client/van_serviceinterface_update_test.go

```diff
@@ -0,0 +1,394 @@
+	cache.WaitForCacheSync(ctx.Done(), svcInformer.HasSynced)
+
+	// create three service targets
+	deployments := cli.KubeClient.AppsV1().Deployments(namespace)
+	statefulSets := cli.KubeClient.AppsV1().StatefulSets(namespace)
+
+	_, err = deployments.Create(tcpDeployment)
+	assert.Assert(t, err)
+	_, err = statefulSets.Create(tcpStatefulSet)
+	assert.Assert(t, err)
+	_, err = deployments.Create(httpDeployment)
+	assert.Assert(t, err)
+
+	err = cli.VanServiceInterfaceCreate(ctx, &types.ServiceInterface{
+		Address:      "noaddress",
+		Protocol:     "tcp",
+		Port:         12345,
+		EventChannel: false,
+		Aggregate:    "",
+	})
+	assert.Error(t, err, "Skupper not initialised in van-serviceinterface-update")
+
+	err = cli.VanRouterCreate(ctx, types.VanSiteConfig{
+		Spec: types.VanSiteConfigSpec{
+			SkupperName:       "skupper",
+			IsEdge:            false,
+			EnableController:  true,
+			EnableServiceSync: true,
+			EnableConsole:     false,
+			AuthMode:          "",
+			User:              "",
+			Password:          "",
+			ClusterLocal:      true,
+		},
+	})
+	assert.Assert(t, err)
+
+	// create three service definitions
+	time.Sleep(time.Second * 1)
+	err = cli.VanServiceInterfaceCreate(ctx, &types.ServiceInterface{
+		Address:      "tcp-go-echo",
+		Protocol:     "tcp",
+		Port:         9090,
+		EventChannel: false,
+		Aggregate:    "",
+	})
+	assert.Assert(t, err)
+
+	err = cli.VanServiceInterfaceCreate(ctx, &types.ServiceInterface{
+		Address:      "nginx",
+		Protocol:     "http",
+		Port:         8080,
+		EventChannel: false,
+		Aggregate:    "",
+	})
+	assert.Assert(t, err)
+
+	err = cli.VanServiceInterfaceCreate(ctx, &types.ServiceInterface{
+		Address:      "tcp-go-echo-ss",
+		Protocol:     "tcp",
+		Port:         9090,
+		EventChannel: false,
+		Aggregate:    "",
+	})
+	assert.Assert(t, err)
+
+	time.Sleep(time.Second * 1)
+
+	// bind services to targets
+	// TODO: could range on list if target type was not needed for bind
+	si, err := cli.VanServiceInterfaceInspect(ctx, "tcp-go-echo")
+	assert.Assert(t, err)
+	err = cli.VanServiceInterfaceBind(ctx, si, "deployment", "tcp-go-echo", "tcp", 9090)
+	assert.Assert(t, err)
+
+	si, err = cli.VanServiceInterfaceInspect(ctx, "tcp-go-echo-ss")
+	assert.Assert(t, err)
+	err = cli.VanServiceInterfaceBind(ctx, si, "statefulset", "tcp-go-echo-ss", "tcp", 9090)
+	assert.Assert(t, err)
+
+	si, err = cli.VanServiceInterfaceInspect(ctx, "nginx")
+	assert.Assert(t, err)
+	// bad bind
+	err = cli.VanServiceInterfaceBind(ctx, si, "deployment", "nginx2", "http", 8080)
+	assert.Error(t, err, "Could not read deployment nginx2: deployments.apps \"nginx2\" not found")
+	// good bind
+	err = cli.VanServiceInterfaceBind(ctx, si, "deployment", "nginx", "http", 8080)
+	assert.Assert(t, err)
+
+	items, err := cli.VanServiceInterfaceList(ctx)
+	assert.Assert(t, err)
+	assert.Equal(t, len(items), 3)
+
+	if *clusterRun {
+		// this delay is for service-controller to update
+		time.Sleep(time.Second * 30)
```

Comment:
similar to previous comment, this sleep may be a global defined variable (or environment variable), "serviceControllerUpdateDelayS" or similar

Reply:
andd would be nice to add some debugging message on every hardcoded sleep, so we can easily track testing hard delays